control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -118,19 +128,48 @@ func findTagReference(tags []imagev1.TagReference, name string) *imagev1.TagRefe
 	return nil
 }
 
-var CatalogToImage = map[string]string{
-	"certified-operators": "registry.redhat.io/redhat/certified-operator-index:v4.16",
-	"community-operators": "registry.redhat.io/redhat/community-operator-index:v4.16",
-	"redhat-marketplace":  "registry.redhat.io/redhat/redhat-marketplace-index:v4.16",
-	"redhat-operators":    "registry.redhat.io/redhat/redhat-operator-index:v4.16",
+func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullSecret []byte) (map[string]string, error) {
+	imageRef := hcp.Spec.ReleaseImage
+	imageConfig, _, _, err := registryclient.GetMetadata(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image metadata: %w", err)
+	}
+
+	versionParts := strings.Split(imageConfig.Config.Labels["io.openshift.release"], ".")
+	if len(versionParts) < 2 {
+		return nil, fmt.Errorf("invalid OpenShift release version format: %s", imageConfig.Config.Labels["io.openshift.release"])
+	}
+
+	operators := map[string]string{
+		"certified-operators": fmt.Sprintf("registry.redhat.io/redhat/certified-operator-index:v%s.%s", versionParts[0], versionParts[1]),
+		"community-operators": fmt.Sprintf("registry.redhat.io/redhat/community-operator-index:v%s.%s", versionParts[0], versionParts[1]),
+		"redhat-marketplace":  fmt.Sprintf("registry.redhat.io/redhat/redhat-marketplace-index:v%s.%s", versionParts[0], versionParts[1]),
+		"redhat-operators":    fmt.Sprintf("registry.redhat.io/redhat/redhat-operator-index:v%s.%s", versionParts[0], versionParts[1]),
+	}
+
+	_, err = registryclient.GetListDigest(ctx, operators["certified-operators"], pullSecret)
+	if err != nil && ctx.Value(TestKey) != "test-reconcile-olm" {
```

Comment:
It is weird to have test-aware code in the function.

Also this whole block seems fragile.  Maybe we could break it out and make it unit testable by itself so the tests document the expected cases.